### PR TITLE
Document reflection free Jackson serializers

### PR DESCRIPTION
**Pull request by mariofusco**
Follow up of https://github.com/quarkusio/quarkus/pull/41063

### CODE REVIEW

File: docs/src/main/asciidoc/rest.adoc [hunk 1]
```diff
@@ -1581,6 +1581,12 @@ public User userPrivate() {
 When the result the `userPublic` method is serialized, the `id` field will not be contained in the response as the `Public` view does not include it.
 The result of `userPrivate` however will include the `id` as expected when serialized.
 
+===== Reflection-free Jackson serialization
+
+Out-of-the-box Jackson serialization converts objects into their json representation introspecting them through a heavy use of reflection. However, the general Quarkus philosophy is to avoid reflection as much as possible, often replacing it with build time code generation. Following this idea, also in this case it is possible to automatically generate at build time implementations of the Jackson `StdSerializer`, one for each class to be converted in json, that can be subsequently used by Quarkus at runtime to perform the json serialization of the objects returned by a rest endpoint without any use of reflection.
```
Comment (gsmet) [inline docs/src/main/asciidoc/rest.adoc:1586]:
```suggestion
Out-of-the-box Jackson serialization converts objects into their JSON representation by introspecting them through a heavy use of reflection. However, the general Quarkus philosophy is to avoid reflection as much as possible, often replacing it with build time code generation. Following this idea, also in this case it is possible to automatically generate at build time implementations of the Jackson `StdSerializer`, one for each class to be converted in JSON, that can be subsequently used by Quarkus at runtime to perform the JSON serialization of the objects returned by a REST endpoint without any use of reflection.
```

File: docs/src/main/asciidoc/rest.adoc [hunk 2]
```diff
@@ -1581,6 +1581,12 @@ public User userPrivate() {
...
+
+This feature is turned off by default, but it can be enabled setting the configuration property `quarkus.rest.jackson.optimization.enable-reflection-free-serializers=true`.
```
Comment (gsmet) [inline docs/src/main/asciidoc/rest.adoc:1588]:
```suggestion
This feature is turned off by default, but it can be enabled by setting the configuration property `quarkus.rest.jackson.optimization.enable-reflection-free-serializers=true`.
```

File: docs/src/main/asciidoc/rest.adoc (hunk 1 above)
Comment (gsmet) [reply docs/src/main/asciidoc/rest.adoc:1586]: The sentence starting with `Following this idea` is very convoluted and it's very hard to follow (at least IMO). Typically the `also in this case` is very confusing to me. Maybe split things in smaller sentences?

**Review by gsmet — CHANGES_REQUESTED** [review]
Thanks! I suggested a few improvements.

**Comment by mariofusco** [comment]
> Thanks! I suggested a few improvements.

Thanks for your suggestions, I merged them and try to simplify and split that convoluted sentence, please give a second look.

**Comment by geoand** [comment]
Can you please squash the commits?

Thanks!

**Review by geoand — APPROVED** [review]

**Review by gsmet — APPROVED** [review]
Perfect, thanks!
